src/platform/BUILD.gn

```diff
@@ -49,7 +49,6 @@ if (chip_device_platform != "none" && chip_device_platform != "external") {
     chip_device_config_enable_wpa = chip_enable_wifi
 
     defines = [
-      "CHIP_DEVICE_CONFIG_ENABLE_THREAD=${chip_enable_openthread}",
```

Comment:
I have removed this define deliberately. 
I think it would be good to have this define set in `CHIPDevicePlatformConfig.h`, because we can use Kconfig defines directly there.

Reply:
I think this would be changing behavior on all CHIP platforms, not just nrfconnect.  @mspang ?

Reply:
It seems fine, that's why there's a change to src/platform/EFR32/CHIPDevicePlatformConfig.h in this PR.